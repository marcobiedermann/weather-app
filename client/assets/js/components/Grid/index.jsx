import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.module.css';

const Grid = (props) => {
  const { children, className, ...otherProps } = props;

  return (
    <div className={classNames(className, styles.grid)} {...otherProps}>
      {children}
    </div>
  );
};

Grid.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

Grid.defaultProps = {
  children: null,
  className: '',
};

export default Grid;
