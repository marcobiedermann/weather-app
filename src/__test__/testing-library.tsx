import { RenderOptions, render } from '@testing-library/react';
import { ReactElement, ReactNode } from 'react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';
import i18n from './i18n';

interface ProvidersProps {
  children: ReactNode;
}

function Providers(props: ProvidersProps) {
  const { children } = props;

  return (
    <I18nextProvider i18n={i18n}>
      <MemoryRouter>{children}</MemoryRouter>
    </I18nextProvider>
  );
}

function customRender(ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) {
  return render(ui, {
    wrapper: Providers,
    ...options,
  });
}

export { customRender as render };
