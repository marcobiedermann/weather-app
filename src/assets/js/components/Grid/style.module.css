.grid {
  margin-left: auto;
  margin-right: auto;
  max-width: 1080px;
  width: 90%;
}
