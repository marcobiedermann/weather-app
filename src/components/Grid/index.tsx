import { ReactNode } from 'react';
import styles from './style.module.css';

interface GridProps {
  children: ReactNode;
}

function Grid(props: GridProps): JSX.Element {
  return <div className={styles.grid} {...props} />;
}

export type { GridProps };
export default Grid;
