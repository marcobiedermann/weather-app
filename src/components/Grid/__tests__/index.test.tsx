import { describe, expect, it } from 'vitest';
import Grid from '..';
import { render } from '../../../__test__/testing-library';

describe('components/Grid', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Grid>Content</Grid>);

    expect(asFragment()).toMatchSnapshot();
  });
});
