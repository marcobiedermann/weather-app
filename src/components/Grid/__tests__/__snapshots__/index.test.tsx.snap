// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Grid renders correctly 1`] = `
<DocumentFragment>
  <div
    class="grid"
  >
    Content
  </div>
</DocumentFragment>
`;
